ci/ray_ci/automation/docker_tags_lib.py

```diff
@@ -169,6 +176,88 @@ def delete_tag(tag: str, docker_hub_token: str) -> bool:
     return True
 
 
+def _is_release_tag(
+    tag: str,
+    release_versions: Optional[List[str]] = None,
+) -> bool:
+    """
+    Check if tag is a release tag & is in the list of release versions.
+    Args:
+        tag: Docker tag name
+        release_versions: List of release versions.
+            If None, don't filter by release version.
+    Returns:
+        True if tag is a release tag and is in the list of release versions.
+            False otherwise.
+    """
+    variables = tag.split(".")
+    if len(variables) != 3 and "post1" not in tag:
+        return False
+    if not variables[0].isnumeric() or not variables[1].isnumeric():
+        return False
+    if (
+        not variables[2].isnumeric()
+        and "rc" not in variables[2]
+        and "-" not in variables[2]
+    ):
+        return False
+
+    if "-" in variables[2]:
+        variables[2] = variables[2].split("-")[0]
+    release_version = ".".join(variables)
+    if release_versions and release_version not in release_versions:
+        return False
+
+    return True
+
+
+def _call_crane_cp(tag: str, source: str, aws_ecr_repo: str):
+    try:
+        with subprocess.Popen(
+            [
+                "crane",
```

Comment:
where does this binary comes from; or this script is only called in an environment that has this binary?

Reply:
This one calls `crane cp` command in terminal as a subprocess. I'm still figuring out a way to convert this to use an API instead but this one is trickier than other crane functions.